CONFIGURATION.md

```diff
@@ -0,0 +1,184 @@
+
+
+## Database and cache configuration
+
+| Variable             | Description                  | Default Value / Environment Variable | Optional |
+|----------------------|------------------------------|--------------------------------------------------|----------|
+| `redis.uri`          | URI for connecting to Redis. | `redis://${REDIS_HOST:redis}:${REDIS_PORT:6379}` | false    |
+| `redis.pool.enabled` | Enable redis pool.           | `true`                                             | true     |
+| `surrealdb.ns`       | Surreal database namespace.  | `${SURREALDB_NS}`                                | false    |
+| `surrealdb.db`       | Surreal database name.       | `${SURREALDB_DB}`                                | false    |
+| `surrealdb.url`      | Surreal database url.        | `${SURREALDB_URL}`                               | false    |
+| `surrealdb.user`     | Surreal database username.   | `${SURREALDB_USER}`                              | false    |
+| `surrealdb.password` | Surreal database password.    | `${SURREALDB_PASSWORD`                               | false    |
+| `surrealdb.init-db`  | Should initiate surreal DB.   |                           | true     |
+
+
+## Email configuration
+
+| Variable                      | Description                                         | Default Value / Environment Variable | Optional |
+|-------------------------------|-----------------------------------------------------|-------------------------------------------|----------|
+| `mail.from`                          | Email address for sending mail from wave.                         |       | false    |
+
+
+## Jackson configuration
+
+| Variable                      | Description                                         | Default Value / Environment Variable | Optional |
+|-------------------------------|-----------------------------------------------------|-------------------------------------------|----------|
+| `jackson.serialization.writeDatesAsTimestamps` | Write dates as timestamps in jackson serialization. | `false`                                  | false    |
+
+
+## Micronaut specific Configuration
+
+| Variable                    | Description                                                                                 | Default Value / Environment Variable | Optional |
+|-----------------------------|---------------------------------------------------------------------------------------------|-------------------------------------------|----------|
+| `micronaut.caches.cache-20sec.expire-after-access` | Cache with a 20-second expire-after-access policy.                                          | `20s`                                     | false    |
+| `micronaut.caches.cache-1min.expire-after-access`  | Cache with a 1-minute expire-after-access policy.                                           | `1m`                                      | false    |
```

Comment:
It's not clear to me what these parameters are used for.

Moreover, can the user define their own using the same pattern? `micronaut.caches.cache-49sec.expire-after-access`

And why are these required (Optional: false) but are not defined in the wave prod config?

The description is also confusing

Reply:
Many of these sonfig are present in application.yml file
prod config contains properties, which are either not defined or needs to be overridden